pkg/webhookadmission/utils.go

```diff
@@ -0,0 +1,141 @@
+package webhookadmission
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"time"
+
+	"github.com/portworx/sched-ops/k8s"
+	log "github.com/sirupsen/logrus"
+	hook "k8s.io/api/admissionregistration/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	webhookName        = "stork.webhook.com"
+	storkService       = "stork-service"
+	storkNamespace     = "kube-system"
+	storkServiceAccess = "stork-service.kube-system.svc"
+)
+
+// CreateMutateWebhook create new webhookconfig for stork if not exist already
+func CreateMutateWebhook(caBundle []byte) error {
+	path := "/mutate"
+
+	webhook := hook.Webhook{
+		Name: webhookName,
+		ClientConfig: hook.WebhookClientConfig{
+			Service: &hook.ServiceReference{
+				Name:      storkService,
+				Namespace: storkNamespace,
+				Path:      &path,
+			},
+			CABundle: caBundle,
+		},
+		Rules: []hook.RuleWithOperations{
+			{
+				Operations: []hook.OperationType{hook.Create},
+				Rule: hook.Rule{
+					APIGroups:   []string{"apps", ""},
+					APIVersions: []string{"v1"},
+					Resources:   []string{"deployments", "statefulset", "pods"},
```

Comment:
Does this need to be singular or plural?

Reply:
work in both cases, but lets keep it plural for all